src/cluster.rs

```diff
@@ -153,12 +154,12 @@ impl ClusterConnection {
     /// block indefinitely. It is an error to pass the zero `Duration` to this
     /// method.
     pub fn set_write_timeout(&self, dur: Option<Duration>) -> RedisResult<()> {
-        let mut t = self.write_timeout.borrow_mut();
-        *t = dur;
         let connections = self.connections.borrow();
         for conn in connections.values() {
             conn.set_write_timeout(dur)?;
         }
+        let mut t = self.write_timeout.borrow_mut();
```

Comment:
Why is this ordering important? Maybe add a comment here explaining this?

Reply:
I've added a check & custom error. I think it is self-explanatory now.

Reply:
Hey, if you look at the updated code, I've moved it back. In case the user passes a zero duration, we would update the local value first, then attempt to set it for each connection which would then raise the error.
So instead of doing that, I had moved the updating local value part after updating it for each connection.
After your comment, I've moved that part back & instead added a direct check for zero duration before updating & also added an error message.